### PR TITLE
Add swi-prolog recipe [skip appveyor]

### PR DESCRIPTION
**Pull request by osallou**
Add the recipe for Prolog interpreter swi-prolog

### CODE REVIEW

File: recipes/swi-prolog/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+requirements:
+  build:
+    - curl
+    - openjdk
+    - autoconf
+    - gcc   # [not osx]
+    - gmp
+    - llvm  # [osx]
+    - libgcc  # [not osx]
+    - ncurses
```
Comment (bgruening) [inline recipes/swi-prolog/meta.yaml]: nnucrese needs to be pinned, same for zlib and gmp I assume

File: recipes/swi-prolog/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  commands:
+    - swipl -h
+
+about:
+  home: https://github.com/SWI-Prolog/swipl-devel
+  license: Simplified BSD
+  summary: A comprehensive Prolog implementation
+
+extra:
+  skip-lints:
```
Comment (bgruening) [inline recipes/swi-prolog/meta.yaml]: this can be remove I think, this only applies to bioconda packages

File: recipes/swi-prolog/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+source:
+  #url: https://github.com/SWI-Prolog/swipl-devel/archive/V7.7.0.tar.gz
+  # Makes use of git_url as it needs submodules from git, not available in archive
+  git_url: https://github.com/SWI-Prolog/swipl-devel.git
+  git_rev: V{{ version }}
+
+build:
+  number: 0
+  skip: true  # [win]
```
Comment (bgruening) [inline recipes/swi-prolog/meta.yaml]: skip: true  # [win or osx]

File: recipes/swi-prolog/build.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+
+##strictly use anaconda build environment
```
Comment (bgruening) [inline recipes/swi-prolog/build.sh]: this can be cleaned up

File: recipes/swi-prolog/meta.yaml (hunk 1 above)
Comment (osallou) [reply recipes/swi-prolog/meta.yaml]: is there any variable for pinning those common libs ? Can't find in conda forge doc for pinning

**Comment by osallou** [comment]
circleci failed with error:

    /bin/bash: ./scripts/run_docker_build.sh: No such file or directory

base image error or circleci error???

**Comment by bgruening** [comment]
@osallou can you merge the master into your branch?

**Comment by cmonjeau** [comment]
I'm very interested in this recipe @osallou :)

**Comment by bgruening** [comment]
@cmonjeau enjoy!

**Pull request merged**